Descending Keys Tests

andytill opened the pull request:
Tests for descending keys.

nickelization committed:
Fix race in repl_rt_heartbeat due to short timeout

One particular timeout in the repl_rt_heartbeat test was slightly too
short, which could cause us to occasionally hit a false positive on this
test if various timings lined up just right. This PR bumps up the
timeout, which should prevent this from happening again.

I would really like to do a proper fix for this, which would use
intercepts or something to confirm that the actual timeout is being hit
in the code...but we don't really have time for that, and a half fix is
better than no fix I suppose.

ksauzz committed:
Add cleanup intercepts function

This function purges specified target module, which was created as a proxy,
and its backuped original module, then reloads original module.

ksauzz committed:
Update spec

lukebakken committed:
Syntax error in .riak_test.config

Commit:
Tests for flexible TS keys.

Commit:
Tests for flexible TS keys.

Commit:
Removed export_all directive.

Commit:
Removed export_all directive.

Commit:
Merge branch 'at-single_key_ops' of https://github.com/basho/riak_test into at/flexi_keys

Commit:
Everything uses spaces for indentation, commented in tests.

Commit:
first cut of common test running under riak test

Commit:
first cut of common test running under riak test

Commit:
First cut of extending riak_test to run common_tests
* auto-detection of common test tests
* encapsulation of output correctly
* single compile cycle

Things missing as yet:
* handing Offset/Worker specifications

Commit:
First cut of extending riak_test to run common_tests
* auto-detection of common test tests
* encapsulation of output correctly
* single compile cycle

Things missing as yet:
* handing Offset/Worker specifications

Commit:
Test for no quantum.

Commit:
Test for no quantum.

Commit:
Tests for relaxing the restriction on the quantum function coming last.

Commit:
Tests for relaxing the restriction on the quantum function coming last.

Commit:
wip

Commit:
wip

Commit:
Move data creation next to the test functions.

Commit:
Move data creation next to the test functions.

Commit:
intermediate commit of new tests

Commit:
intermediate commit of new tests

Commit:
intermediate commit of new tests

Commit:
fix up merge

Commit:
Added an assertion for row sets that does diffs and ct's for queries around quantum boundaries.

Commit:
Tests for start/end keys around quantum boundaries with all operators.

Commit:
Merge branch 'riak_ts-develop' of https://github.com/basho/riak_test into feature_common_test

Conflicts:
	rebar.config

Commit:
Merge branch 'feature_common_test' of https://github.com/basho/riak_test into at/flexi_keys

Commit:
Rename cluster keys.

Commit:
Converted to common tests.

Commit:
Added null tests for additional local key fields.

Commit:
Added null tests for additional local key fields.

Commit:
More error case checking, some quanta spanning tests.

Commit:
Tests for requiring the paritition key to be covered in the where clause.

Commit:
Tests for requiring the paritition key to be covered in the where clause.

Commit:
Tests for where filters on local key additional fields.

Commit:
Tests for booleans in keys.

Commit:
Moved boolean table creation into one function.

Commit:
Test for different keys and types, and around the edges of quanta.

Commit:
Table definitions with time stamps before and after the quantum.

Commit:
Merge branch 'at/flexi_keys' of https://github.com/basho/riak_test into desc-timestamp-1.3

Commit:
Need tdiff in the escript, or you waste hours working out why.

Commit:
Merge branch 'at/flexi_keys' of https://github.com/basho/riak_test into desc-timestamp-1.3

Commit:
Descending keys tests.

Commit:
Descending keys tests.

Commit:
Descending keys tests.

Commit:
Merge branch 'riak_ts-develop' of https://github.com/basho/riak_test into desc-timestamp-1.3

Conflicts:
	.gitignore
	rebar.config
	src/riak_test_escript.erl
	tests/ts_cluster_keys_SUITE.erl
	tests/ts_cluster_quantum_boundaries_SUITE.erl
	tests/ts_simple_single_key_ops.erl

Commit:
Converted the describe table test into a ct suite.

Commit:
Converted the describe table test into a ct suite.

Commit:
Test for describing a table with a descending key.

Commit:
Test for describing a table with a descending key.

Commit:
Added a descending keys test for the HTTP API.

Commit:
Added a descending keys test for the HTTP API.

Commit:
Fix single op writes.

Commit:
Merge branch 'riak_ts-develop' of https://github.com/basho/riak_test into desc-timestamp-1.3

Commit:
Fixed the list keys tests to use the new client API. Changed the data written to the table with additional local fields to show DESC keys.

Commit:
Fixed the list keys tests to use the new client API. Changed the data written to the table with additional local fields to show DESC keys.

hazen commented on tests/ts_updown_select_SUITE.erl:
```diff
@@ -0,0 +1,78 @@
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+-module(ts_updown_select_SUITE).
```
Since this is a cluster, I would suggest `ts_cluster_updown_select_SUITE`

hazen replied:
Yup. Still think that's a more consistent name

Commit:
Merge branch 'riak_ts-develop' of https://github.com/basho/riak_test into desc-timestamp-1.3

Conflicts:
	tests/ts_simple_describe_table.erl

macintux committed:
Restructure code from ts_util.erl to make tests more explicit/more verbose/much easier to understand

macintux committed:
Convert one simple test

Commit:
Merge branch 'riak_ts-develop' of github.com:basho/riak_test into desc-timestamp-1.3

hmmr committed:
new module to check query buffer operations

hmmr committed:
use {allow_qbuf_reuse, true} in query/5 Options, instead of ONLY keyword

hmmr committed:
ts_cluster_coverage: no MaxQuantum any more

hmmr committed:
query_buffers: milliseconds in timestamps

also, extend time range.

Commit:
Descending keys tests change, set DESC on the timestamp.

Commit:
Descending keys tests change, set DESC on the timestamp.

Commit:
common testify! descending keys select tests.

Commit:
common testify! descending keys select tests.

Commit:
rt for descending key varchars.

Commit:
rt for descending key varchars.

hmmr committed:
ts_simple_query_buffers: proofread and add more comments

hmmr committed:
ts_simple_query_buffers_ldb_errors: new module

hmmr committed:
query buffers: new tests for query_too_big and ldb errors

hmmr committed:
Merge branch 'riak_ts-develop' into feature-az-query_buffers

hmmr committed:
ts_simple_qbuf_ldb: reset writing perms on .../query_buffers dir

hmmr committed:
ts_simple_query_buffers: move error handling tests to main module

with a new test added to check for max query quanta error condition.

Commit:
Lock the dependencies as a baseline

Commit:
Test that bucket types can be modified before activation and continue to work.

Commit:
Test that bucket types can be modified before activation and continue to work.

Commit:
Lock deps for riak_ee-2.2.0rc10ex1

Commit:
added Status field to SQL "SHOW TABLES" result.

hmmr committed:
rt_client: correctly work with both stagedevrel and devrel

hmmr committed:
ts_simple_query_buffers: the default query_buffers path has changed

with some function renames

hmmr committed:
ts_updown_util: hooks to modify riak.conf for curr/prev versions

hmmr committed:
ts_updown_util: honour use_previous_client in CREATE, INSERT queries

hmmr committed:
whitespace cleanup

hmmr committed:
ts_updown_util: ct:log cluster node versions as they are before CREATE

hmmr committed:
rt_client: supply correct paths

generally, make the bloody thing work for the first time.

hmmr committed:
ts_updown_util: correctly emulate rt:pbc on previous client node

hmmr committed:
ts_cluster_updowngrade_group_by: expect all scenarios vs queries to pass

hmmr committed:
ts_updown_util: report cluster composition before CREATE and before SELECT

fadushin committed:
Modified the CRDT test to account for changes we have made in Yokozuna.

This change requires a change to the intercept code, as we refactored the delete operations in Yokozuna to avoid deleteByQuery, and hence has different touch points in Yokozuna that need to be intercepted.

hmmr committed:
ts_updown_util: use wait_until_capability before issuing queries

This is to ensure that on node transition, any capabilities change
take effect across all nodes.

hmmr committed:
new module ts_cluster_updowngrade_order_by_SUITE

macintux committed:
Faster/cleaner `remove_last/1` per Andrei

converting ts_util calls to new ts library calls

modified ts_setup:create_bucket_type

more test updating

Replace ts_util with ts_setup, ts_ops and ts_data

Commit:
Rename intercept to match updated code in yokozuna.

russelldb committed:
Add gset support to DT tests

Commit:
Merge branch 'riak_ts-develop' of github.com:basho/riak_test into desc-timestamp-1.3

Commit:
Make the describe suite assert against columns instead of rows, so when a row is added, it doesn't break all the tests.

Commit:
Make the describe suite assert against columns instead of rows, so when a row is added, it doesn't break all the tests.

gordonguthrie commented:
+1 6681d5d

hmmr committed:
ts_simple_http_security_SUITE: rework, make it actually pass

hmmr committed:
Merge branch 'riak_ts-develop' into feature-az-query_buffers

hazen commented:
@andytill Why do we have so many changes on this?  Was this some very old version which you started from? Maybe we could rebase your changes?

korry8911 committed:
added the command line ability to pass in tests as comma seperated values after -t

Commit:
Merge pull request #1187 from basho/kc-ts_util-update

Break ts_util into three utility modules

Commit:
Lock deps for riak_ts_ee-1.5.0alpha4

Commit:
Merge pull request #1185 from basho/fix-az-unbreak_ts_simple_http_security_suite

ts_simple_http_security_SUITE: rework, make it actually pass

hmmr committed:
ts_updown_util: rework to halt scenario if a stage fails

hmmr committed:
ts_cluster_updowngrade_*: exclude downgrades from scenarios, for 1.5-1.4

hmmr committed:
ts_updown_util: more checkpoint messages in logs

hmmr committed:
ts_cluster_updowngrade_: reenable previous QueryNodeVsn (exc. order_by)

korry8911 committed:
support passing in comma separated list of wildcards

korry8911 committed:
made it better

hmmr committed:
ts_updown_util: teach find_or_make_node_at_vsn to properly select nodes

hmmr committed:
ts_updown_util: more logging messages

hmmr committed:
ts_cluster_updowngrade_*: enable more scenarios, add new caps to check

this requires tag riak_ts-1.4.1 in riak_kv

korry8911 committed:
made the script more efficient

korry8911 committed:
one more change

Commit:
Merge pull request #1186 from basho/kc-csv-cl-options

Pass in tests to riak_test script as comma separated values

Commit:
Prevent tree upgrades which can block replication from completing

Use capability override to make sure that neither side
of the replication link upgrades its object hash version
because if once side updates its trees and the other hasn't
completed the process yet (or in mixed clusters doesn't even
support it) the AAE full sync cannot complete.

ian-mi committed:
Use rt:wait_until to check unacked on source node

Acks are sent asynchronously after a successful write on the sink, so we
cannot rely on the ack having being received by the source immediately
after we read the data on the sink.

bsparrow435 committed:
Merge pull request #1188 from basho/dr-force_hashtree_upgrade_on_replication_object_reformat

Prevent tree upgrades which can block replication from completing

JeetKunDoug committed:
Merge pull request #1189 from basho/im/bugfix/rt_cascading_ensure_ack-race

Use rt:wait_until to check unacked on source node

Commit:
Add rt:wait_until to verify counter

Added a rt:wait_until for the final check of the counters instead of
only checking once. Sometimes, even though the full sync completed
successfully, the counter values might not be fully written and
propagated to all owning vnodes. Change test to check more than once to
account for this eventual consistency.

JeetKunDoug committed:
Merge pull request #1190 from basho/bjs-bugfix-wait_until-verify-counter-repl

Add rt:wait_until to final counter check

fadushin committed:
Removed superfluous call repair after being cancelled.

Commit:
Add additional logging to sibling debug log in object_value.

Needed to help figure out why certain repl tests have failed on
occasion.

JeetKunDoug committed:
Merge pull request #1192 from basho/fd-verify_2i_aae

Removed superfluous call repair after being cancelled.

JeetKunDoug committed:
Merge pull request #1191 from basho/dr-additional_logging_for_repl_custom_n

Add additional logging to sibling debug log in object_value. [JIRA: RIAK-2971]

Commit:
Merge remote-tracking branch 'origin/develop-2.2' into riak_ts-develop

Commit:
Merge pull request #1152 from basho/feature-az-query_buffers

new module to check query buffer operations

hazen commented on run_tests.sh:
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+./riak_test -c rt_ts -t tests/ts_updown_select_SUITE.erl
```
This seems pretty specific.  Was this meant to be included in this PR?

hazen commented on src/riak_test_escript.erl:
```diff
@@ -149,6 +149,7 @@ main(Args) ->
                     Tests0;
                 {_, undefined} ->
                     Tests0;
+                %% TODO don't understand what this does
```
Did you want to add this comment?  I can explain (more) of what this is meant to do.  When `riak_test` was written it was supposed to run a whole bunch of tests sequentially and this shuffles the order.  Not that useful now that we have independent tests on giddyup.  It was meant to break any potential test interdependence.

andytill replied:
Not sure where that is from. I need to rebase this, it will probably disappear then.

gordonguthrie replied:
I left it in there - it was something to do with parallel tests running in the cloud - but I don't understand why/how it is used...

hazen commented on tests/ts_cluster_ddl_upgrades_SUITE.erl:
```diff
@@ -0,0 +1,323 @@
+    [{timetrap,{minutes,10}}].
+
+init_per_suite(Config) ->
+    % ct:pal("VERSION 1.4 ~p", [rt:find_version_by_name(["riak_ts-1.4.0", "riak_ts_ee-1.4.0"])]),
```
Mean to leave this in?

hazen reviewed (changes requested):
Let me know your thoughts on my comments. Thanks!

russelldb committed:
Merge pull request #1184 from basho/rdb/bet365/gsets

Add support for gsets

Commit:
Merge remote-tracking branch 'origin/develop-2.2' into riak_ts-develop

Commit:
Lock deps for riak_ee-2.2.0rc16

Commit:
Merge pull request #1176 from basho/jbg_show_tables_active_status

Add Status field to SQL "SHOW TABLES" result

Commit:
Merge branch 'desc-timestamp-1.3' of github.com:basho/riak_test into desc-timestamp-1.3

hazen commented:
Think we'll need a rebase on this PR since I doubt we actually have 103 relevant file changes...

andytill commented:
@javajolt I will close this one and open another PR tomorrow, something went very wrong with the rebase I did today.

andytill commented:
Closing and reopening as https://github.com/basho/riak_test/pull/1200
